core/src/main/java/org/elasticsearch/cluster/ClusterInfo.java

```diff
@@ -19,23 +19,23 @@
 
 package org.elasticsearch.cluster;
 
-import com.google.common.collect.ImmutableMap;
+import org.elasticsearch.cluster.routing.AllocationId;
 import org.elasticsearch.cluster.routing.ShardRouting;
 
+import java.util.Collection;
+import java.util.Collections;
 import java.util.Map;
 
 /**
  * ClusterInfo is an object representing a map of nodes to {@link DiskUsage}
- * and a map of shard ids to shard sizes, see
- * <code>InternalClusterInfoService.shardIdentifierFromRouting(String)</code>
- * for the key used in the shardSizes map
+ * and a map of shard ids to shard sizes keyed by the {@link AllocationId}
  */
 public class ClusterInfo {
-
+    public static final ClusterInfo EMPTY = new ClusterInfo(Collections.EMPTY_MAP, Collections.EMPTY_MAP);
```

Comment:
I _think_ (Collections.<String, DiskUsage>emptyMap(), Collection.<AllocationId, Long>emptyMap()) is more "java" here even (and maybe especially) if it is a horrible mouth full.